packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -523,6 +523,66 @@ export class TestObjectProvider implements ITestObjectProvider {
 	}
 }
 
+export class TestObjectProviderWithVersionedLoad extends TestObjectProvider {
+	constructor(
+		public readonly LoaderConstructor: typeof Loader,
+		public readonly LoaderConstructorForLoading: typeof Loader,
```

Comment:
nits, and could we also clarify the purpose of these loaders? The names are a bit confusing
```suggestion
		public readonly loaderConstructor: typeof Loader,
		public readonly loaderConstructorForLoading: typeof Loader,
```